usdQt/hierarchyModel.py

```diff
@@ -0,0 +1,457 @@
+        """Return the predicate used to identify prims that should be exposed to the model
+        """
+        return self.__index.GetPredicate()
+
+    def GetRoot(self):
+        """Retrieve the root of the current hierarchy model"""
+        rootProxy = self.__index.GetRoot()
+        return rootProxy.GetPrim() if rootProxy and not rootProxy.expired else None
+
+    def __OnObjectsChanged(self, notice, sender):
+        resyncedPaths = notice.GetResyncedPaths()
+        resyncedPaths = [path for path in resyncedPaths if path.IsPrimPath()]
+
+        if len(resyncedPaths) > 0:
+            with self.__LayoutChangedContext(self):
+                persistentIndices = self.persistentIndexList()
+                indexToPath = {}
+                for index in persistentIndices:
+                    indexProxy = index.internalPointer()
+                    indexPrim = indexProxy.GetPrim()
+                    indexPath = indexPrim.GetPath()
+
+                    for resyncedPath in resyncedPaths:
+                        commonPath = resyncedPath.GetCommonPrefix(path)
+                        # if the paths are siblings or if the
+                        # index path is a child of resynced path, you need to update
+                        # any persistent indices
+                        areSiblings = commonPath == resyncedPath.GetParentPath() and commonPath != indexPath
+                        indexIsChild = (commonPath == resyncedPath)
+
+                        if areSiblings or indexIsChild:
+                            indexToPath[index] = indexPath
+
+                self.__index.ResyncSubtrees(resyncedPaths)
+
+                fromIndices = []
+                toIndices = []
+                for index in indexToPath:
+                    path = indexToPath[index]
+
+                    if self.__index.ContainsPath(path):
+                        newProxy = self.__index.GetProxy(path)
+                        newRow = self.__index.GetRow(newProxy)
+
+                        if (index.row() != newRow):
+                            for i in xrange(self.columnCount(QtCore.QModelIndex())):
+                                fromIndices.append(index)
+                                toIndices.append(self.createIndex(
+                                    newRow, index.column(), newProxy))
+                    else:
+                        fromIndices.append(index)
+                        toIndices.append(QtCore.QModelIndex())
+                self.changePersistentIndexList(fromIndices, toIndices)
+
+    def _GetPrimForIndex(self, modelIndex):
+        """Retrieve the prim for the input modelIndex
+
+        External clients should use UsdQt.roles.HierarchyPrimRole to access
+        the prim for an index"""
+        if modelIndex.isValid():
+            proxy = modelIndex.internalPointer()
+            proxy = compatability.ResolveValue(proxy)
+            if type(proxy) == _HierarchyCache.Proxy and not proxy.expired:
```

Comment:
Fixed.  I don't know if we want subclasses of the Proxies just yet.  I think we'll have to go down this path at some point... but not quite yet  (For example, I expect we'll want to allow for specialization of handling of things like Primvars which might be spanned across multiple attributes if they are indexed).